### DISCUSSION TITLE
Cabling documentation

### DISCUSSION THREAD

**Discussion by MvQuast** [Neuroquila-n8fall/JunkersControl#4] [Q&A] [answered]
Hi Dave,

Thanks for providing this project. I'm about to get this implemented for my Junkers ZBS 16/170.
Do you have any further guidance on cabling between ESP32, Nieren board and BM1 on the heating itself?

Thanks 
Marcel

**Comment by Neuroquila-n8fall** [comment, marked as answer]
The cabling is documented on the manual of the BM1. If you're looking for power for your ESP32 be advised that if you don't have a proper converter on board you have to use an external power supply or converter in general because voltages on the BM1 range between 17 and 24VDC:
Here you have it:
![image](https://user-images.githubusercontent.com/29146258/137072572-fa2e0a3c-fa75-4a0b-9302-d9e57d92a190.png)

If using the code as-is using NiRen module and ESP32 the wiring needs to be as follows:
![image](https://user-images.githubusercontent.com/29146258/137075795-a9c5e01f-9ddf-4203-a29c-7b05e5d6d603.png)

The configuration is also somewhat flexible and can be modified to your needs.
See https://github.com/Neuroquila-n8fall/JunkersControl/blob/5a476f19802a7795b31d9acf4021a6e28a0f910e/include/can_processor.h#L13
and
https://github.com/Neuroquila-n8fall/JunkersControl/blob/5a476f19802a7795b31d9acf4021a6e28a0f910e/include/can_processor.h#L14


NOTE: Although CAN works on 5V signal level I found that powering the transceiver using 5V will result in received frames but the inability to send. This shouldn't be an issue - in fact it should be mandatory to power the transceiver (TJA1050) using 5V and the interface (MCP2515) using logic level of the uC (3v3 in the case of an ESP32). So it is absolutely safe and properly working when powering the NiRen module using 3v3. **Don't ever try to power it with 5V since this raises the logic level to 5V and damages the ESP32 permanently.**
